File: ledger-python/ledger/report_download_service.py
"""Serves previously generated report files back to the caller.

Reports live in a shared spool directory written by the nightly
export job and the ad-hoc export endpoint, and are cleaned up after
7 days by a separate retention job.
"""

import os

REPORT_SPOOL_DIR = "/var/fusion-demo/report-spool"


def load_report_bytes(report_file_name):
    report_path = os.path.join(REPORT_SPOOL_DIR, report_file_name)
    with open(report_path, "rb") as report_file:
        return report_file.read()

File: gateway-java/src/main/java/com/fusiondemo/gateway/reports/ReportDownloadService.java
package com.fusiondemo.gateway.reports;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;

/**
 * Serves previously generated report files (PDF/CSV exports) back to
 * the requesting client. Reports are written to a shared spool
 * directory by the nightly export job and cleaned up after 7 days.
 */
public class ReportDownloadService {

    private static final String REPORT_SPOOL_DIR = "/var/fusion-demo/report-spool";

    public byte[] loadReport(String reportFileName) {
        File file = new File(REPORT_SPOOL_DIR, reportFileName);
        try {
            return Files.readAllBytes(file.toPath());
        } catch (IOException e) {
            throw new RuntimeException("Unable to read report file: " + reportFileName, e);
        }
    }
}
